pkgs/top-level/all-packages.nix

```diff
@@ -10715,14 +10715,14 @@ in
   rust_1_45 = callPackage ../development/compilers/rust/1_45.nix {
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
   };
-  rust_1_49 = callPackage ../development/compilers/rust/1_49.nix {
+  rust_1_50 = callPackage ../development/compilers/rust/1_50.nix {
```

Comment:
Why get rid of 1.49?

Reply:
We usually only expose the latest rustc compiler.

Also, with rust, there's no immediate reason to keep older versions, as rust has a solid backwards compatibility model.

Reply:
Well, in theory at least, looks like there's a regression above.

Reply:
Gotcha... But it looks like 1.45 has some special standing?

Reply:
according to git history, it's related to musl https://github.com/NixOS/nixpkgs/issues/94228

Reply:
> > Firefox no longer builds on this branch:
> > ```
> > 
> > error: failed to parse manifest at `/build/firefox-85.0.1/Cargo.toml`
> > 
> > 
> > 
> > Caused by:
> > 
> >   dependency (nix) specification is ambiguous. Only one of `branch`, `tag` or `rev` is allowed.
> > ```
> 
> Sounds like that's a breaking change from cargo 1.50.0. what should we do about it? Wait for a new Firefox, or patch the Firefox cargo toml/lock

Either patch it or keep an old version around for firefox.